src/Element.cc

```diff
@@ -181,8 +231,33 @@ void Element::AddAttribute(const std::string &_key,
       this->CreateParam(_key, _type, _defaultValue, _required, _description));
 }
 
+/////////////////////////////////////////////////
+void Element::AddAttribute(const std::string &_key,
+                           const std::string &_type,
+                           const std::string &_defaultValue,
+                           bool _required,
+                           sdf::Errors &_errors,
+                           const std::string &_description)
+{
+  this->dataPtr->attributes.push_back(
```

Comment:
This is duplicating code. See my comments on `AddValue`.

Reply:
Yes, fixed.